src/runtime_src/core/tools/xbtracer/src/app/launcher.cpp

```diff
@@ -118,6 +139,65 @@ void log_d(const Args&... args)
   }
 }
 
+//bool set_env(const std::string &key, const std::string &value)
+bool set_env(const char* key, const char* value)
+{
+#ifdef _WIN32
+  errno_t err = _putenv_s(key, value);
+  return err == 0;
+#else
+  int ret = setenv(key, value, 1);
+  return ret == 0;
+#endif
+}
+
+std::string get_env(std::string key)
+{
+#if _WIN32
+  char *val = nullptr;
+  size_t len = 0;
+  errno_t err = _dupenv_s(&val, &len, key.c_str());
+  if (err != 0 || val == nullptr)
+    return std::string();
+  std::string result(val);
+  free(val); // Free the memory allocated by _dupenv_s
```

Comment:
This is not exception safe. std::string ctor may throw.   This _dupenv_s business is very unfortunate, what does MSFT recommend in order to make this exception safe?   I think you have to create a unique_ptr with val immediately after _dupenv_s , the unique_ptr should call `free` when going out of scope.   Horrible mess :-(

Reply:
Done, please check if it looks alright.